app/portainer/react/components/custom-templates/variables-field.ts

```diff
@@ -33,6 +33,9 @@ class VariablesFieldController implements IComponentController {
   }
 
   $onChanges({ value }: IOnChangesObject) {
+    if (value === undefined) {
+      return;
+    }
```

Comment:
can value be null? note that `value` is bad naming (on my and angular's side). it's not the actual value but its changes object.

this will take care of `undefined` and `null`
```suggestion
    if (value?.currentValue) {
      this.checkValidity(value.currentValue);
    }
```